ocaml/utils/language_extension.ml

```diff
@@ -201,3 +227,75 @@ let disallow_extensions () =
   (* The strictest option, so no internal checks needed *)
   extensions := [];
   universe   := Universe.No_extensions
+
+(* converts full extension names, like "layouts_alpha" to a pair of
+   an extension and its setting *)
+let pair_of_string extn_name : extn_pair option =
+  match String.lowercase_ascii extn_name with
+  | "comprehensions" -> Some (Pair (Comprehensions, ()))
+  | "local" -> Some (Pair (Local, ()))
+  | "include_functor" -> Some (Pair (Include_functor, ()))
+  | "polymorphic_parameters" -> Some (Pair (Polymorphic_parameters, ()))
+  | "immutable_arrays" -> Some (Pair (Immutable_arrays, ()))
+  | "strengthening" -> Some (Pair (Module_strengthening, ()))
+  | "layouts" -> Some (Pair (Layouts, (Stable : Maturity.t)))
+  | "layouts_beta" -> Some (Pair (Layouts, (Beta : Maturity.t)))
+  | "layouts_alpha" -> Some (Pair (Layouts, (Alpha : Maturity.t)))
+  | _ -> None
+
+let pair_of_string_exn extn_name = match pair_of_string extn_name with
+  | Some pair -> pair
+  | None ->
+    raise (Arg.Bad(Printf.sprintf "Extension %s is not known" extn_name))
+
+let of_string extn_name =
+  let pack (Pair (extn, _) : extn_pair) = Pack extn in
+  Option.map pack (pair_of_string extn_name)
+
+let enable_of_string_exn extn_name = match pair_of_string_exn extn_name with
+  | Pair (extn, setting) -> enable extn setting
+
+let disable_of_string_exn extn_name = match pair_of_string_exn extn_name with
+  | Pair (extn, _) -> disable extn
+
+module Exist = struct
+  type 'a extn = 'a t
+  type t = exist =
+    | Pack : 'a extn -> t
+
+  let to_command_line_strings (Pack extn) =
+    let (module Ops) = get_level_ops extn in
+    List.map
+      (fun level -> to_string extn ^ Ops.to_command_line_suffix level)
+      Ops.all
+
+  let to_string : t -> string = function
+    | Pack extn -> to_string extn
+
+  let is_enabled : t -> bool = function
+    | Pack extn -> is_enabled extn
+
+  let is_erasable : t -> bool = function
+    | Pack extn -> is_erasable extn
+
+  let all : t list =
```

Comment:
It seems easy to forget to add something to `all`.

A low-tech way of making this less likely would be to move the definition of `all` up higher in the file, perhaps closer to one of the exhaustive matches like `equal_t` or `to_string`. (It seems fine to me to e.g. define `Pack` and `all` at the top-level of the ml file and then re-export it, and `all`, from `Exist`.)